src/KReduce.chpl

```diff
@@ -67,6 +67,7 @@ module KReduce {
     }
 
     proc generate() {
+      v.doSort();
```

Comment:
Is this necessary in the common case?  It seems like it will add overhead.  I'd be inclined to move the .doSort() into any "accumulate-after-combine" branches since we expect that to be rare and so aren't as worried about the performance there (or else, have that branch set a flag that a doSort() is still required in order to do it here).

Reply:
This is only needed when doing serial reductions. Arkouda doesn't do this, but I ran into this for some standalone reproducers. `doSort` is a no-op if things are already sorted, so there's no real overhead.

Reply:
Oh, actually that's a lie, there is no check in `doSort` the `isSorted` check is in the merge code.

Reply:
Is this ready to be merged?

Reply:
do we have a test case that exhibits the previous bug?